phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4957,12 +4964,22 @@ private void returnAllSequences(ConcurrentMap<SequenceKey,Sequence> sequenceMap)
     public void addConnection(PhoenixConnection connection) throws SQLException {
         if (returnSequenceValues || shouldThrottleNumConnections) {
             synchronized (connectionCountLock) {
-                if (shouldThrottleNumConnections && connectionCount + 1 > maxConnectionsAllowed){
-                    GLOBAL_PHOENIX_CONNECTIONS_THROTTLED_COUNTER.increment();
-                    throw new SQLExceptionInfo.Builder(SQLExceptionCode.NEW_CONNECTION_THROTTLED).
-                        build().buildException();
+
+                if(shouldThrottleNumConnections) {
+                    int futureConnections = 1 + ( connection.isInternalConnection() ? internalConnectionCount : connectionCount);
+                    int allowedConnections = connection.isInternalConnection() ? maxInternalConnectionsAllowed : maxConnectionsAllowed;
+                    if(allowedConnections != 0 && futureConnections > allowedConnections) {
+                        GLOBAL_PHOENIX_CONNECTIONS_THROTTLED_COUNTER.increment();
```

Comment:
Better to add a comment somewhere around here about the expected working of counting internal phoenix conns and how/if this works against client conns, etc.

Reply:
Sure